modules/planning/lattice/trajectory_generator/trajectory_evaluator.cc

```diff
@@ -0,0 +1,258 @@
+    : is_auto_tuning_(is_auto_tuning),
+      pathtime_neighborhood_(pathtime_neighborhood) {
+  for (const auto lon_trajectory : lon_trajectories) {
+    if (!LatticeConstraintChecker::IsValidLongitudinalTrajectory(
+            *lon_trajectory)) {
+      continue;
+    }
+    for (const auto lat_trajectory : lat_trajectories) {
+      if (!LatticeConstraintChecker::IsValidLateralTrajectory(
+              *lat_trajectory, *lon_trajectory)) {
+        continue;
+      }
+      if (!is_auto_tuning_) {
+        double cost =
+            evaluate(planning_target, lon_trajectory, lat_trajectory, nullptr);
+        cost_queue_.push(PairCost({lon_trajectory, lat_trajectory}, cost));
+      } else {
+        std::vector<double> cost_components;
+        double cost = evaluate(planning_target, lon_trajectory, lat_trajectory,
+                               &cost_components);
+        cost_queue_with_components_.push(PairCostWithComponents(
+            {lon_trajectory, lat_trajectory}, {cost_components, cost}));
+      }
+    }
+  }
+  if (!is_auto_tuning_) {
+    AINFO << "Number of valid 1d trajectory pairs:\t" << cost_queue_.size();
+  } else {
+    AINFO << "Number of valid 1d trajectory pairs:\t"
+          << cost_queue_with_components_.size();
+  }
+}
+
+bool TrajectoryEvaluator::has_more_trajectory_pairs() const {
+  if (!is_auto_tuning_) {
+    return !cost_queue_.empty();
+  } else {
+    return !cost_queue_with_components_.empty();
+  }
+}
+
+std::size_t TrajectoryEvaluator::num_of_trajectory_pairs() const {
+  if (!is_auto_tuning_) {
+    return cost_queue_.size();
+  } else {
+    return cost_queue_with_components_.empty();
+  }
+}
+
+std::pair<std::shared_ptr<Trajectory1d>, std::shared_ptr<Trajectory1d>>
+TrajectoryEvaluator::next_top_trajectory_pair() {
+  CHECK(has_more_trajectory_pairs() == true);
+  if (!is_auto_tuning_) {
+    auto top = cost_queue_.top();
+    cost_queue_.pop();
+    return top.first;
+  } else {
+    auto top = cost_queue_with_components_.top();
+    cost_queue_with_components_.pop();
+    return top.first;
+  }
+}
+
+double TrajectoryEvaluator::top_trajectory_pair_cost() const {
+  if (!is_auto_tuning_) {
+    return cost_queue_.top().second;
+  } else {
+    return cost_queue_with_components_.top().second.second;
+  }
+}
+
+std::vector<double> TrajectoryEvaluator::top_trajectory_pair_component_cost()
+    const {
+  CHECK(is_auto_tuning_);
+  return cost_queue_with_components_.top().second.first;
+}
+
+double TrajectoryEvaluator::evaluate(
+    const PlanningTarget& planning_target,
+    const std::shared_ptr<Trajectory1d>& lon_trajectory,
+    const std::shared_ptr<Trajectory1d>& lat_trajectory,
+    std::vector<double>* cost_components) const {
+  // currently consider three costs:
+  // 1. the cost of jerk, currently only consider longitudinal jerk.
+  // 2. the cost of time to achieve the objective
+  // 3. the cost of failure to achieve the planning objective.
+
+  double weight_lon_travel = 1000.0;
```

Comment:
So does the weight, maybe put them in param configs in next iteration

Reply:
Yes, we will move the constants to gflags in next iterations.